Sources/ArcGISToolkit/Components/OfflineMapAreasView/OfflineMapAreasView.MapViewModel.swift

```diff
@@ -0,0 +1,212 @@
+        /// offline map models.
+        func makePreplannedOfflineMapModels() async {
+            preplannedMapModels = await Result {
+                try await offlineMapTask.preplannedMapAreas
+                    .sorted(using: KeyPathComparator(\.portalItem.title))
+                    .compactMap {
+                        PreplannedMapModel(
+                            preplannedMapArea: $0,
+                            offlineMapTask: self.offlineMapTask,
+                            preplannedDirectory: self.preplannedDirectory
+                        )
+                    }
+            }
+            if let models = try? preplannedMapModels?.get() {
+                hasPreplannedMapAreas = !models.isEmpty
+            }
+        }
+        
+        /// Handles a selection of a map.
+        /// If the selected map is an offline map and it has not yet been taken offline, then
+        /// it will start downloading. Otherwise the selected map will be used as the displayed map.
+        func selectedMapDidChange(from oldValue: SelectedMap) {
+            switch selectedMap {
+            case .onlineWebMap:
+                offlineMap = nil
+            case .preplannedMap(let info):
+                if info.canDownload {
+                    // If we have not yet downloaded or started downloading, then kick off a
+                    // download and reset selection to previous selection since we have to download
+                    // the offline map.
+                    selectedMap = oldValue
+                    Task {
+                        await info.downloadPreplannedMapArea()
+                        loadPreplannedMobileMapPackages()
+                    }
+                } else if case .success(let mmpk) = info.result {
+                    // If we have already downloaded, then open the map in the mmpk.
+                    offlineMap = mmpk.maps.first
+                } else {
+                    // If we have a failure, then keep the online map selected.
+                    selectedMap = oldValue
+                }
+            }
+        }
+        
+        /// Loads locally stored mobile map packages for the map's preplanned map areas.
+        func loadPreplannedMobileMapPackages() {
+            // Create mobile map packages with saved mmpk files.
+            let mmpkFiles = searchFiles(in: preplannedDirectory, with: "mmpk")
+            self.mobileMapPackages.removeAll()
+            
+            for fileURL in mmpkFiles {
+                let mobileMapPackage = MobileMapPackage(fileURL: fileURL)
+                self.mobileMapPackages.append(mobileMapPackage)
+            }
+            
+            // Pass mobile map packages to preplanned map models.
+            if let models = try? preplannedMapModels?.get() {
+                models.forEach { model in
+                    model.setMobileMapPackages(mobileMapPackages)
```

Comment:
This seems like it's setting all the mobile map packages on each preplanned map model. Shouldn't it only set the ones that are relevant to that particular preplanned map model?